### ISSUE TITLE
Sierra install issues

### ISSUE THREAD

**Issue by matthewcosgrove** [geerlingguy/ansible-role-homebrew#54]
Fresh install of Sierra 10.12.2 and this ansible role installs Homebrew 1.1.0 but it errors and fails at the following point

```
TASK [geerlingguy.homebrew : Force update brew after installation.] ************
fatal: [127.0.0.1]: FAILED! => {"changed": true, "cmd": ["/usr/local/bin/brew", "update", "--force"],
"delta": "0:00:00.147456", "end": "2016-12-27 16:02:43.209803", "failed": true, "rc": 1, "start": "2016-12-27 16:02:43.062347", 
"stderr": "Error: Running Homebrew as root is extremely dangerous and no longer supported.\n
As Homebrew does not drop privileges on installation you would be giving all\n
build scripts full access to your system.", 
"stdout": "", "stdout_lines": [], "warnings": []}

PLAY RECAP *********************************************************************
127.0.0.1                  : ok=9    changed=5    unreachable=0    failed=1 
```
To unblock progressing the role to allow completion I'm thinking I should just disable the specific "Force update" task which could be done via this approach http://allwrightythen.com/using-a-forked-ansible-galaxy-role/

Could someone please clarify the best way forwards to workaround and/or fix this?

Thanks,

Matthew

p.s. for reference the previous commands run were

```
Matthews-MacBook-Air:macos-bootstrap matthewcosgrove$ sudo ansible-galaxy install -r pre-req-roles.yml
- downloading role 'dotfiles', owned by geerlingguy
- downloading role from https://github.com/geerlingguy/ansible-role-dotfiles/archive/1.1.0.tar.gz
- extracting geerlingguy.dotfiles to /etc/ansible/roles/geerlingguy.dotfiles
- geerlingguy.dotfiles was installed successfully
- downloading role 'homebrew', owned by geerlingguy
- downloading role from https://github.com/geerlingguy/ansible-role-homebrew/archive/2.1.0.tar.gz
- extracting geerlingguy.homebrew to /etc/ansible/roles/geerlingguy.homebrew
- geerlingguy.homebrew was installed successfully
Matthews-MacBook-Air:macos-bootstrap matthewcosgrove$ ansible-playbook main.yml -i inventory -K
Traceback (most recent call last):
  File "/usr/local/bin/ansible-playbook", line 43, in <module>
    import ansible.constants as C
  File "/Library/Python/2.7/site-packages/ansible/constants.py", line 161, in <module>
    DEFAULT_LOCAL_TMP         = get_config(p, DEFAULTS, 'local_tmp',        'ANSIBLE_LOCAL_TEMP',      '$HOME/.ansible/tmp', istmppath=True)
  File "/Library/Python/2.7/site-packages/ansible/constants.py", line 81, in get_config
    makedirs_safe(value, 0o700)
  File "/Library/Python/2.7/site-packages/ansible/utils/path.py", line 67, in makedirs_safe
    raise AnsibleError("Unable to create local directories(%s): %s" % (to_native(rpath), to_native(e)))
ansible.errors.AnsibleError: Unable to create local directories(/Users/matthewcosgrove/.ansible/tmp): [Errno 13] Permission denied: '/Users/matthewcosgrove/.ansible/tmp'
Matthews-MacBook-Air:macos-bootstrap matthewcosgrove$ sudo !!
sudo ansible-playbook main.yml -i inventory -K
```

**Comment by matthewcosgrove** [comment]
Got further by running

```
sudo chown -R $USER:admin /usr/local
brew update --force
```

but more suggestions would be very welcome

**Comment by geerlingguy** [comment]
@matthewcosgrove - Are you using a downstream project that uses this role? The `sudo ansible-galaxy install -r pre-req-roles.yml` command isn't in any of this role's documentation, nor does my [mac-dev-playbook](https://github.com/geerlingguy/mac-dev-playbook) use it.

Homebrew shouldn't be installed with sudo, and the role would/should fail if being run as your current user without the `/usr/local` directory permissions set correctly...

**Comment by matthewcosgrove** [comment]
@geerlingguy ah yes, I meant put the link to https://github.com/matthewcosgrove/macos-bootstrap which is effectively a stripped down version of your playbook but which I started from scratch.

And I appreciate that sudo should not be used for brew however as seen in the dump in my p.s. above I only added sudo to get over another issue. Excerpt is

```
Matthews-MacBook-Air:macos-bootstrap matthewcosgrove$ ansible-playbook main.yml -i inventory -K
Traceback (most recent call last):
  File "/usr/local/bin/ansible-playbook", line 43, in <module>
    import ansible.constants as C
  File "/Library/Python/2.7/site-packages/ansible/constants.py", line 161, in <module>
    DEFAULT_LOCAL_TMP         = get_config(p, DEFAULTS, 'local_tmp',        'ANSIBLE_LOCAL_TEMP',      '$HOME/.ansible/tmp', istmppath=True)
  File "/Library/Python/2.7/site-packages/ansible/constants.py", line 81, in get_config
    makedirs_safe(value, 0o700)
  File "/Library/Python/2.7/site-packages/ansible/utils/path.py", line 67, in makedirs_safe
    raise AnsibleError("Unable to create local directories(%s): %s" % (to_native(rpath), to_native(e)))
ansible.errors.AnsibleError: Unable to create local directories(/Users/matthewcosgrove/.ansible/tmp): [Errno 13] Permission denied: '/Users/matthewcosgrove/.ansible/tmp'
Matthews-MacBook-Air:macos-bootstrap matthewcosgrove$ sudo !!
```

To clarify my position, the steps on https://github.com/matthewcosgrove/macos-bootstrap worked fine on a VirtualBox VM using El Capitan but when applied to Sierra on a clean Macbook Air install I came across the issues above which prevented me from progressing. I am currently creating a Sierra Virtual Box to reproduce the issue on another machine, but if you have any thoughts it would be appreciated.

**Comment by matthewcosgrove** [comment]
@geerlingguy Works fine on the VirtualBox VM and comparing the two I see that after running my ./install-ansible.sh script it created the .ansible directory owned by root on the Macbook Air but owned by my user on the VirtualBox VM.

It's not clear to me how that can be the case as I have run through the same set of steps and scripts for both, but anyway I will close this issue.
